test/test_synthetic.py

```diff
@@ -0,0 +1,407 @@
+            if low_ht < ht:
+                low_ht = ht
+
+            # Continue only if needed - 1m troposphere does nothing
+            if np.abs(high_ht - low_ht) < 1.0:
+                continue
+
+            low_xyz = getTopOfAtmosphere(xyz, LOS, low_ht, factor=cos_factor)
+            high_xyz = getTopOfAtmosphere(xyz, LOS, high_ht, factor=cos_factor)
+
+            ray_length = np.linalg.norm(high_xyz - low_xyz, axis=-1)
+
+            outSubs   += ray_length
+
+            cos_factor = (high_ht - low_ht) / ray_length if cos_factor is None else cos_factor
+
+    return outputArrs
+
+
+@dataclass
+class StudyArea(object):
+    """ Object with shared parameters related to the study area """
+    reg:str
+    WM:str
+    wd = op.join(TEST_DIR, 'synthetic_test')
+
+    def __post_init__(self):
+        self.setup_region()
+
+        self.dts   = self.dt.strftime('%Y_%m_%d_T%H_%M_%S')
+        self.ttime = self.dt.strftime('%H:%M:%S')
+
+        self.wmObj = modelName2Module(self.WM.upper().replace("-", ""))[1]()
+
+        aoi = BoundingBox(self.SNWE)
+        aoi.add_buffer(buffer = 1.5 * self.wmObj.getLLRes())
+
+        self.los  = Raytracing(self.orbit, time=self.dt)
+
+        wm_bounds = aoi.calc_buffer_ray(self.los.getSensorDirection(), lookDir=self.los.getLookDirection())
+        self.wmObj.set_latlon_bounds(wm_bounds)
+        wm_fname  = make_weather_model_filename(self.WM, self.dt, self.wmObj._ll_bounds)
+        self.path_wm_real = op.join(self.wd, 'weather_files_real', wm_fname)
+
+        grid_spacing = 0.5
+        self.cube_spacing = np.round(grid_spacing/1e-5).astype(np.float32)
+        self.hgt_lvls     = np.arange(-500, 1500, 500)
+
+
+    def setup_region(self):
+        if self.reg == 'LA':
+            self.SNWE  = 33, 34, -118.25, -117.25
+            self.dt    = datetime(2020, 1, 30, 13, 52, 45)
+            self.orbit = self.wd + \
+                '/S1B_OPER_AUX_POEORB_OPOD_20210317T025713_V20200129T225942_20200131T005942.EOF'
+
+        elif self.reg == 'Fort':
+            self.SNWE = -4.0, -3.5, -38.75, -38.25
+            self.dt   = datetime(2019, 11, 17, 20, 51, 58)
+            self.orbit = self.wd + \
+                '/S1A_OPER_AUX_POEORB_OPOD_20210315T014833_V20191116T225942_20191118T005942.EOF'
+
+
+    def make_config_dict(self):
+        dct = {
+            'aoi_group': {'bounding_box': list(self.SNWE)},
+            'height_group': {'height_levels': self.hgt_lvls.tolist()},
+            'time_group': {'time': self.ttime, 'interpolate_time': False},
+            'date_group': {'date_list': datetime.strftime(self.dt, '%Y%m%d')},
+            'cube_spacing_in_m': str(self.cube_spacing),
+            'los_group': {'ray_trace': True, 'orbit_file': self.orbit},
+            'weather_model': self.WM,
+            'runtime_group': {'output_directory': self.wd},
+        }
+        return dct
+
+
+def dl_real(reg='LA', mod='GMAO'):
```

Comment:
All the things above are technically "fixtures". Would be nice to use the pytest decorator as such, but not required.

https://docs.pytest.org/en/6.2.x/fixture.html